File: frontend/src/types/nda.ts
export interface PartyData {
  signature: string; // base64 data URL
  printName: string;
  title: string;
  company: string;
  noticeAddress: string;
  date: string;
}

export interface NdaFormData {
  purpose: string;
  effectiveDate: string;
  mndaTermType: "expires" | "perpetual";
  mndaTermYears: number | null;
  confidentialityTermType: "years" | "perpetual";
  confidentialityTermYears: number | null;
  governingLawState: string;
  jurisdiction: string;
  modifications: string;
  party1: PartyData;
  party2: PartyData;
}

export const defaultParty: PartyData = {
  signature: "",
  printName: "",
  title: "",
  company: "",
  noticeAddress: "",
  date: "",
};

export const defaultFormData: NdaFormData = {
  purpose:
    "Evaluating whether to enter into a business relationship with the other party.",
  effectiveDate: new Date().toISOString().split("T")[0],
  mndaTermType: "expires",
  mndaTermYears: 1,
  confidentialityTermType: "years",
  confidentialityTermYears: 1,
  governingLawState: "",
  jurisdiction: "",
  modifications: "",
  party1: { ...defaultParty },
  party2: { ...defaultParty },
};

// Keys per wizard step — used for per-step validation trigger
export const STEP_FIELDS: Array<string[]> = [
  ["purpose", "effectiveDate"],
  ["mndaTermType", "mndaTermYears", "confidentialityTermType", "confidentialityTermYears"],
  [
    "party1.printName", "party1.title", "party1.company",
    "party1.noticeAddress", "party1.date",
    "party2.printName", "party2.title", "party2.company",
    "party2.noticeAddress", "party2.date",
    "governingLawState", "jurisdiction",
  ],
  ["party1.signature", "party2.signature"],
  [],
];

export const STEP_TITLES = [
  "Agreement Basics",
  "Duration & Confidentiality",
  "Party Details",
  "Signatures",
  "Review & Download",
];

File: backend/models/nda.py
from typing import Literal, Optional
from pydantic import BaseModel, field_validator, model_validator


class PartyData(BaseModel):
    signature: str
    print_name: str
    title: str
    company: str
    notice_address: str
    date: str

    @field_validator("signature")
    @classmethod
    def signature_must_be_data_url(cls, v: str) -> str:
        if not v:
            raise ValueError("signature is required")
        if not v.startswith("data:image/"):
            raise ValueError("signature must be a base64 image data URL")
        return v


class NdaRequest(BaseModel):
    purpose: str
    effective_date: str
    mnda_term_type: Literal["expires", "perpetual"]
    mnda_term_years: Optional[int] = None
    confidentiality_term_type: Literal["years", "perpetual"]
    confidentiality_term_years: Optional[int] = None
    governing_law_state: str
    jurisdiction: str
    modifications: Optional[str] = None
    party1: PartyData
    party2: PartyData

    @model_validator(mode="after")
    def validate_conditional_years(self) -> "NdaRequest":
        if self.mnda_term_type == "expires" and not self.mnda_term_years:
            raise ValueError("mnda_term_years required when mnda_term_type is 'expires'")
        if self.confidentiality_term_type == "years" and not self.confidentiality_term_years:
            raise ValueError(
                "confidentiality_term_years required when confidentiality_term_type is 'years'"
            )
        return self


class NdaResponse(BaseModel):
    html: str
